File: backend/Mover.Entities/POCOEntities/Vehiculos/Referencia.cs

namespace Mover.Entities.POCOEntities.Vehiculos
{
    public class Referencia
    {
        public int Id { get; set; }
        public string Nombre { get; set; } = string.Empty;
        public string? Descripcion { get; set; } 

        public int? MarcaId { get; set; }
        public Marca? Marca { get; set; }
    }
}

File: frontend/lib/domain/entities/referencia.dart
class Referencia {
  final int? id;
  final String nombre;
  final String? descripcion;
  final int marcaId;

  Referencia(
      {this.id,
      required this.nombre,
      required this.descripcion,
      required this.marcaId});

  factory Referencia.fromJson(Map<String, dynamic> json) {
    return Referencia(
        id: json['id'],
        nombre: json['nombre'],
        descripcion: json['descripcion'],
        marcaId: json['marcaId']);
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'descripcion': descripcion,
      'marcaId': marcaId
    };
  }
}
